Extension fails to upload the binary: uninitialized constant Bundler::Plugin::API::Source

``` 
2017-03-01T10:16:33.5277550Z /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.5/lib/bundler/rubygems_ext.rb:45:in `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)
 
2017-03-01T10:16:33.5296750Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
 
2017-03-01T10:16:33.5315480Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
 
2017-03-01T10:16:33.5334100Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
 
2017-03-01T10:16:33.5352480Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
 
2017-03-01T10:16:33.5370880Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
 
2017-03-01T10:16:33.5389290Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
 
2017-03-01T10:16:33.5407680Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
 
2017-03-01T10:16:33.5426050Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.5/lib/bundler/plugin/api/source.rb:2:in `<top (required)>'
 
2017-03-01T10:16:33.5444130Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.5/lib/bundler/rubygems_ext.rb:45:in `full_gem_path'
 
2017-03-01T10:16:33.5462500Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
 
2017-03-01T10:16:33.5481660Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
 
2017-03-01T10:16:33.5500590Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
 
2017-03-01T10:16:33.5520720Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
 
2017-03-01T10:16:33.5539300Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
 
2017-03-01T10:16:33.5557530Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
 
2017-03-01T10:16:33.5575910Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
 
2017-03-01T10:16:33.5594320Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.5/lib/bundler.rb:12:in `<top (required)>'
 
2017-03-01T10:16:33.5612460Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
 
2017-03-01T10:16:33.5630020Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
 
2017-03-01T10:16:33.5647260Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane/plugins/plugin_manager.rb:5:in `<class:PluginManager>'
 
2017-03-01T10:16:33.5664430Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane/plugins/plugin_manager.rb:4:in `<module:Fastlane>'
 
2017-03-01T10:16:33.5681560Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane/plugins/plugin_manager.rb:3:in `<top (required)>'
 
2017-03-01T10:16:33.5699110Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5716450Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5733580Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane/plugins/plugins.rb:9:in `<top (required)>'
 
2017-03-01T10:16:33.5750690Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5767910Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5784900Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane.rb:22:in `<top (required)>'
 
2017-03-01T10:16:33.5803320Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5821590Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
 
2017-03-01T10:16:33.5839570Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
 
2017-03-01T10:16:33.5856370Z 	from /Users/builder/.gem-cache/gems/fastlane-2.18.3/bin/fastlane:15:in `<top (required)>'
 
2017-03-01T10:16:33.5873030Z 	from /Users/builder/.gem-cache/bin/fastlane:23:in `load'
 
2017-03-01T10:16:33.5889360Z 	from /Users/builder/.gem-cache/bin/fastlane:23:in `<main>'
 
2017-03-01T10:16:33.5903860Z ##[debug]rc:1
 
2017-03-01T10:16:33.5916340Z ##[debug]success:false
 
2017-03-01T10:16:33.5928830Z ##[debug]task result: Failed
 
2017-03-01T10:16:33.6013110Z ##[error]Error: fastlane failed with return code: 1
 
2017-03-01T10:16:33.6026010Z ##[debug]Processed: ##vso[task.issue type=error;]Error: fastlane failed with return code: 1
 
2017-03-01T10:16:33.6047820Z ##[debug]Processed: ##vso[task.complete result=Failed;]Error: fastlane failed with return code: 1
```

I've tried with different bundler versions:
bundler-1.14.5, bundler-1.14.3

I can report the same issue with my App Store Build task, using only the task defaults. It's grabbing the bad bundler version during the fastlane gem install, so this will probably affect everybody until the aforementioned fix is released. I couldn't find a way to override the bundler/fastlane version -- is there one? 

```
2017-03-02T22:13:07.7524690Z ##[section]Starting: Publish to the App Store TestFlight track
2017-03-02T22:13:08.0512830Z d16d4221-d33b-4eff-ae89-809a91fe4f37 exists true
2017-03-02T22:13:08.0619620Z [command]/usr/bin/gem install fastlane
2017-03-02T22:13:46.5361960Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for frameit/lib/assets/empty.png, skipping
2017-03-02T22:13:55.8501590Z Successfully installed fastlane-2.19.1
2017-03-02T22:13:55.8516460Z Parsing documentation for fastlane-2.19.1
2017-03-02T22:13:55.8529230Z 1 gem installed
2017-03-02T22:13:55.9247000Z [command]/usr/bin/gem update fastlane -i /Users/vso112771/.gem-cache
2017-03-02T22:13:58.8054300Z Updating installed gems
2017-03-02T22:13:58.8060890Z Nothing to update
2017-03-02T22:13:58.8699580Z [command]fastlane pilot upload -u ******** -i /Users/vso112771/vsts-agent/_work/3/s/bin/ios/release/Conseris.ipa --skip_waiting_for_build_processing true
2017-03-02T22:13:59.9453820Z /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/rubygems_ext.rb:45:in `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)
2017-03-02T22:13:59.9470990Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
2017-03-02T22:13:59.9489790Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
2017-03-02T22:13:59.9505180Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
2017-03-02T22:13:59.9519740Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-02T22:13:59.9536050Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-02T22:13:59.9548330Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
2017-03-02T22:13:59.9557400Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
2017-03-02T22:13:59.9566280Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/plugin/api/source.rb:2:in `<top (required)>'
2017-03-02T22:13:59.9575000Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/rubygems_ext.rb:45:in `full_gem_path'
2017-03-02T22:13:59.9583900Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
2017-03-02T22:13:59.9592950Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
2017-03-02T22:13:59.9602500Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
2017-03-02T22:13:59.9611290Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-02T22:13:59.9620070Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-02T22:13:59.9634070Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
2017-03-02T22:13:59.9643310Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
2017-03-02T22:13:59.9652060Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler.rb:12:in `<top (required)>'
2017-03-02T22:13:59.9660760Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
2017-03-02T22:13:59.9669670Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
2017-03-02T22:13:59.9678490Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:5:in `<class:PluginManager>'
2017-03-02T22:13:59.9687820Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:4:in `<module:Fastlane>'
2017-03-02T22:13:59.9699990Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:3:in `<top (required)>'
2017-03-02T22:13:59.9708990Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9717930Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9726780Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugins.rb:9:in `<top (required)>'
2017-03-02T22:13:59.9796140Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9805190Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9813830Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane.rb:22:in `<top (required)>'
2017-03-02T22:13:59.9823180Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9832070Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-02T22:13:59.9841490Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
2017-03-02T22:13:59.9850560Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/bin/fastlane:15:in `<top (required)>'
2017-03-02T22:13:59.9858740Z 	from /usr/local/bin/fastlane:23:in `load'
2017-03-02T22:13:59.9866710Z 	from /usr/local/bin/fastlane:23:in `<main>'
2017-03-02T22:14:00.0181780Z ##[error]Error: fastlane failed with return code: 1
2017-03-02T22:14:00.1086000Z ##[section]Finishing: Publish to the App Store TestFlight track
```

The manual version selection works like a charm! 

Unfortunately, I couldn't get it to solve this particular issue, as the bad bundler version (1.14.4) is still being used no matter which fastlane version is selected. I even tried going all the way back to fastlane 1.92.0, which was the most recent version that doesn't reference bundler (at least as a runtime dependency), and it still invoked bundler 1.14.4 (and this is even though 1.14.6 is the latest released version). 

This is using a VSTS hosted build agent -- is there a cached toolkit on those maybe? I was trying to figure out how to run "gem install bundler" prior to the fastlane task to see if it might update to 1.14.6, but no luck so far. 

```
2017-03-04T01:31:20.7460760Z ##[section]Starting: Publish to the App Store TestFlight track
2017-03-04T01:31:21.0514520Z d16d4221-d33b-4eff-ae89-809a91fe4f37 exists true
2017-03-04T01:31:21.0639840Z [command]/usr/bin/gem install fastlane -v 1.92.0
2017-03-04T01:32:56.0346460Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/GridExampleScreenshot.png, skipping
2017-03-04T01:32:56.0361960Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/ipadair.png, skipping
2017-03-04T01:32:56.0377420Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/iphone4s.png, skipping
2017-03-04T01:32:56.0392760Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/iphone5s.png, skipping
2017-03-04T01:32:56.0408160Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/iphone6s.png, skipping
2017-03-04T01:32:56.0423580Z unable to convert "\x89" from ASCII-8BIT to UTF-8 for lib/fastlane/actions/device_grid/assets/iphone6splus.png, skipping
2017-03-04T01:33:03.1883120Z Successfully installed spaceship-0.39.0
2017-03-04T01:33:03.1896690Z Successfully installed xcode-install-1.4.0
2017-03-04T01:33:03.1910230Z Successfully installed fastlane-1.92.0
2017-03-04T01:33:03.1923350Z Parsing documentation for spaceship-0.39.0
2017-03-04T01:33:03.1936190Z Installing ri documentation for spaceship-0.39.0
2017-03-04T01:33:03.1948960Z Parsing documentation for xcode-install-1.4.0
2017-03-04T01:33:03.1961810Z Installing ri documentation for xcode-install-1.4.0
2017-03-04T01:33:03.1974570Z Parsing documentation for fastlane-1.92.0
2017-03-04T01:33:03.1990460Z Installing ri documentation for fastlane-1.92.0
2017-03-04T01:33:03.2003940Z 3 gems installed
2017-03-04T01:33:03.2726320Z [command]fastlane pilot upload -u ******** -i /Users/vso112771/vsts-agent/_work/3/s/bin/ios/release/Conseris.ipa --skip_waiting_for_build_processing true
2017-03-04T01:33:04.4365540Z /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/rubygems_ext.rb:45:in `full_gem_path': uninitialized constant Bundler::Plugin::API::Source (NameError)
2017-03-04T01:33:04.4375240Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
2017-03-04T01:33:04.4384440Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
2017-03-04T01:33:04.4393500Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
2017-03-04T01:33:04.4402500Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-04T01:33:04.4411840Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-04T01:33:04.4420840Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
2017-03-04T01:33:04.4433100Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
2017-03-04T01:33:04.4447670Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/plugin/api/source.rb:2:in `<top (required)>'
2017-03-04T01:33:04.4460090Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler/rubygems_ext.rb:45:in `full_gem_path'
2017-03-04T01:33:04.4469080Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1391:in `contains_requirable_file?'
2017-03-04T01:33:04.4478050Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:68:in `block in require'
2017-03-04T01:33:04.4486950Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:823:in `block in each'
2017-03-04T01:33:04.4496140Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-04T01:33:04.4504870Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:822:in `each'
2017-03-04T01:33:04.4513630Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `find'
2017-03-04T01:33:04.4528120Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:67:in `require'
2017-03-04T01:33:04.4542670Z 	from /Library/Ruby/Gems/2.0.0/gems/bundler-1.14.4/lib/bundler.rb:12:in `<top (required)>'
2017-03-04T01:33:04.4557160Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
2017-03-04T01:33:04.4571910Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `require'
2017-03-04T01:33:04.4586720Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:5:in `<class:PluginManager>'
2017-03-04T01:33:04.4601450Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:4:in `<module:Fastlane>'
2017-03-04T01:33:04.4615750Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugin_manager.rb:3:in `<top (required)>'
2017-03-04T01:33:04.4629990Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4645330Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4660650Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/plugins/plugins.rb:9:in `<top (required)>'
2017-03-04T01:33:04.4675390Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4689990Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4704730Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane.rb:22:in `<top (required)>'
2017-03-04T01:33:04.4719290Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4734250Z 	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:73:in `require'
2017-03-04T01:33:04.4749050Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/fastlane/lib/fastlane/cli_tools_distributor.rb:18:in `take_off'
2017-03-04T01:33:04.4763610Z 	from /Users/vso112771/.gem-cache/gems/fastlane-2.19.1/bin/fastlane:15:in `<top (required)>'
2017-03-04T01:33:04.4777480Z 	from /usr/local/bin/fastlane:23:in `load'
2017-03-04T01:33:04.4791060Z 	from /usr/local/bin/fastlane:23:in `<main>'
2017-03-04T01:33:04.4953410Z ##[error]Error: fastlane failed with return code: 1
2017-03-04T01:33:04.5267700Z ##[section]Finishing: Publish to the App Store TestFlight track
```

--- a/README.md
+++ b/README.md
@@ -135,7 +135,7 @@ Allows you to promote an app previously updated to iTunes Connect to the App Sto
 
 4. **fastlane Version** - **Latest Version** or **Specific Version**.  If *Specific Version* is chosen, you must provide a value for *fastlane Specific Version*.
 
-5. **fastlane Specific Version** *(String)* - The version of fastlane to install (e.g., 2.15.1).
+5. **fastlane Specific Version** *(String)* - The version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.
 
 ## Firewall Issues
 

--- a/Tasks/app-store-promote/Strings/resources.resjson/en-US/resources.resjson
+++ b/Tasks/app-store-promote/Strings/resources.resjson/en-US/resources.resjson
@@ -27,7 +27,7 @@
   "loc.input.label.fastlaneToolsVersion": "fastlane Version",
   "loc.input.help.fastlaneToolsVersion": "Choose to install either the lastest version of fastlane or a specific version.",
   "loc.input.label.fastlaneToolsSpecificVersion": "fastlane Specific Version",
-  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).",
+  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
   "loc.messages.DarwinOnly": "The Apple App Store Promote task can only run on a Mac computer.",
   "loc.messages.SuccessfullySubmitted": "Build successfully submitted for review."
 }

--- a/Tasks/app-store-promote/Tests/L0.ts
+++ b/Tasks/app-store-promote/Tests/L0.ts
@@ -53,7 +53,7 @@ describe('app-store-promote L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.invokedToolCount === 1, 'should have only run fastlane pilot.');
+        assert(tr.invokedToolCount === 1, 'should have only run fastlane deliver.');
         assert(tr.succeeded, 'task should have succeeded');
 
         done();
@@ -66,8 +66,9 @@ describe('app-store-promote L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
+        assert(tr.ran('/usr/bin/gem uninstall fastlane -a -I'), 'gem uninstall fastlane should have been run.');
         assert(tr.ran('/usr/bin/gem install fastlane -v 2.15.1'), 'gem install fastlane with a specific version should have been run.');
-        assert(tr.invokedToolCount === 2, 'should have run only gem install and fastlane pilot.');
+        assert(tr.invokedToolCount === 3, 'should have run gem uninstall, gem install and fastlane deliver.');
         assert(tr.succeeded, 'task should have succeeded');
 
         done();
@@ -162,7 +163,7 @@ describe('app-store-promote L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -q teamId --force'), 'fastlane deliver with team id should have been run.');
+        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -k teamId --force'), 'fastlane deliver with team id should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');
@@ -177,7 +178,7 @@ describe('app-store-promote L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -r teamName --force'), 'fastlane deliver with team name should have been run.');
+        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -e teamName --force'), 'fastlane deliver with team name should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');
@@ -192,7 +193,7 @@ describe('app-store-promote L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -q teamId -r teamName --force'), 'fastlane deliver with team id and team name should have been run.');
+        assert(tr.ran('fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -k teamId -e teamName --force'), 'fastlane deliver with team id and team name should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');

--- a/Tasks/app-store-promote/Tests/L0SpecificFastlaneInstall.ts
+++ b/Tasks/app-store-promote/Tests/L0SpecificFastlaneInstall.ts
@@ -38,6 +38,10 @@ let myAnswers: string = `{
         "/usr/bin/fastlane": true
     },
     "exec": {
+        "/usr/bin/gem uninstall fastlane -a -I": {
+            "code": 0,
+            "stdout": "Successfully uninstalled fastlane-2.15.1"
+        },
         "/usr/bin/gem install fastlane -v 2.15.1": {
             "code": 0,
             "stdout": "1 gem installed"

--- a/Tasks/app-store-promote/Tests/L0TeamId.ts
+++ b/Tasks/app-store-promote/Tests/L0TeamId.ts
@@ -47,7 +47,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -q teamId --force": {
+        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -k teamId --force": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-promote/Tests/L0TeamIdTeamName.ts
+++ b/Tasks/app-store-promote/Tests/L0TeamIdTeamName.ts
@@ -48,7 +48,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -q teamId -r teamName --force": {
+        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -k teamId -e teamName --force": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-promote/Tests/L0TeamName.ts
+++ b/Tasks/app-store-promote/Tests/L0TeamName.ts
@@ -47,7 +47,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -r teamName --force": {
+        "fastlane deliver submit_build -u creds-username -a com.microsoft.test.appId --skip_binary_upload true --skip_metadata true --skip_screenshots true -e teamName --force": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-promote/app-store-promote.ts
+++ b/Tasks/app-store-promote/app-store-promote.ts
@@ -72,6 +72,15 @@ async function run() {
         tl.debug('Checking for ruby install...');
         tl.which('ruby', true);
 
+        //Whenever a specific version of fastlane is requested, we're going to uninstall all installed
+        //versions of fastlane beforehand.  Note that this doesn't uninstall dependencies of fastlane.
+        if (installFastlane && fastlaneVersionToInstall) {
+            let gemRunner: ToolRunner = tl.tool(tl.which('gem', true));
+            gemRunner.arg(['uninstall', 'fastlane']);
+            tl.debug(`Uninstalling all fastlane versions...`);
+            gemRunner.arg(['-a', '-I']);  //uninstall all versions
+            await gemRunner.exec();
+        }
         // If desired, install the fastlane tools (if they're already present, should be a no-op)
         if (installFastlane) {
             tl.debug('Installing fastlane...');
@@ -95,7 +104,7 @@ async function run() {
         }
 
         //Run the deliver command 
-        // See https://github.com/fastlane/deliver for more information on these arguments
+        // See https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/options.rb for more information on these arguments
         let deliverCommand: ToolRunner = tl.tool('fastlane');
         deliverCommand.arg(['deliver', 'submit_build', '-u', credentials.username, '-a', appIdentifier]);
         if (chooseBuild.toLowerCase() === 'specify') {
@@ -103,8 +112,8 @@ async function run() {
         }
         deliverCommand.arg(['--skip_binary_upload', 'true', '--skip_metadata', 'true', '--skip_screenshots', 'true']);
         deliverCommand.argIf(shouldAutoRelease, '--automatic_release');
-        deliverCommand.argIf(teamId, ['-q', teamId]);
-        deliverCommand.argIf(teamName, ['-r', teamName]);
+        deliverCommand.argIf(teamId, ['-k', teamId]);
+        deliverCommand.argIf(teamName, ['-e', teamName]);
         deliverCommand.arg('--force');
 
         await deliverCommand.exec();

--- a/Tasks/app-store-promote/task.json
+++ b/Tasks/app-store-promote/task.json
@@ -149,7 +149,7 @@
             "defaultValue": "",
             "required": true,
             "groupName": "advanced",
-            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).",
+            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
             "visibleRule": "fastlaneToolsVersion = SpecificVersion"
         }
     ],

--- a/Tasks/app-store-promote/task.json
+++ b/Tasks/app-store-promote/task.json
@@ -13,7 +13,7 @@
     "demands": [ "xcode" ],
     "version": {
         "Major": "1",
-        "Minor": "114",
+        "Minor": "115",
         "Patch": "0"
     },
     "minimumAgentVersion": "1.95.3",

--- a/Tasks/app-store-promote/task.loc.json
+++ b/Tasks/app-store-promote/task.loc.json
@@ -15,7 +15,7 @@
   ],
   "version": {
     "Major": "1",
-    "Minor": "114",
+    "Minor": "115",
     "Patch": "0"
   },
   "minimumAgentVersion": "1.95.3",

--- a/Tasks/app-store-release/Strings/resources.resjson/en-US/resources.resjson
+++ b/Tasks/app-store-release/Strings/resources.resjson/en-US/resources.resjson
@@ -46,7 +46,7 @@
   "loc.input.label.fastlaneToolsVersion": "fastlane Version",
   "loc.input.help.fastlaneToolsVersion": "Choose to install either the lastest version of fastlane or a specific version.",
   "loc.input.label.fastlaneToolsSpecificVersion": "fastlane Specific Version",
-  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).",
+  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
   "loc.messages.DarwinOnly": "The Apple App Store Release task can only run on a Mac computer.",
   "loc.messages.SuccessfullyPublished": "Successfully published to %s",
   "loc.messages.NoIpaFilesFound": "No IPA file found using pattern: %s",

--- a/Tasks/app-store-release/Tests/L0.ts
+++ b/Tasks/app-store-release/Tests/L0.ts
@@ -121,8 +121,9 @@ describe('app-store-release L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
+        assert(tr.ran('/usr/bin/gem uninstall fastlane -a -I'), 'gem uninstall fastlane should have been run.');
         assert(tr.ran('/usr/bin/gem install fastlane -v 2.15.1'), 'gem install fastlane with a specific version should have been run.');
-        assert(tr.invokedToolCount === 2, 'should have run only gem install and fastlane pilot.');
+        assert(tr.invokedToolCount === 3, 'should have run gem uninstall, gem install and fastlane pilot.');
         assert(tr.succeeded, 'task should have succeeded');
 
         done();
@@ -335,7 +336,7 @@ describe('app-store-release L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -q teamId'), 'fastlane deliver with teamId should have been run.');
+        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -k teamId'), 'fastlane deliver with teamId should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');
@@ -350,7 +351,7 @@ describe('app-store-release L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -r teamName'), 'fastlane deliver with teamName should have been run.');
+        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -e teamName'), 'fastlane deliver with teamName should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');
@@ -365,7 +366,7 @@ describe('app-store-release L0 Suite', function () {
         let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
 
         tr.run();
-        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -q teamId -r teamName'), 'fastlane deliver with teamId and teamName should have been run.');
+        assert(tr.ran('fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -k teamId -e teamName'), 'fastlane deliver with teamId and teamName should have been run.');
         assert(tr.invokedToolCount === 3, 'should have run gem install, gem update and fastlane deliver.');
         assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');

--- a/Tasks/app-store-release/Tests/L0ProductionTeamId.ts
+++ b/Tasks/app-store-release/Tests/L0ProductionTeamId.ts
@@ -53,7 +53,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -q teamId": {
+        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -k teamId": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-release/Tests/L0ProductionTeamIdTeamName.ts
+++ b/Tasks/app-store-release/Tests/L0ProductionTeamIdTeamName.ts
@@ -54,7 +54,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -q teamId -r teamName": {
+        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -k teamId -e teamName": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-release/Tests/L0ProductionTeamName.ts
+++ b/Tasks/app-store-release/Tests/L0ProductionTeamName.ts
@@ -53,7 +53,7 @@ let myAnswers: string = `{
             "code": 0,
             "stdout": "1 gem installed"
         },
-        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -r teamName": {
+        "fastlane deliver --force -u creds-username -a com.microsoft.test.appId -i mypackage.ipa --skip_metadata true --skip_screenshots true -e teamName": {
             "code": 0,
             "stdout": "consider it delivered!"
         }

--- a/Tasks/app-store-release/Tests/L0TestFlightSpecificFastlaneInstall.ts
+++ b/Tasks/app-store-release/Tests/L0TestFlightSpecificFastlaneInstall.ts
@@ -42,6 +42,10 @@ let myAnswers: string = `{
         ]
     },
     "exec": {
+        "/usr/bin/gem uninstall fastlane -a -I": {
+            "code": 0,
+            "stdout": "Successfully uninstalled fastlane-2.15.1"
+        },
         "/usr/bin/gem install fastlane -v 2.15.1": {
             "code": 0,
             "stdout": "1 gem installed"

--- a/Tasks/app-store-release/app-store-release.ts
+++ b/Tasks/app-store-release/app-store-release.ts
@@ -103,6 +103,15 @@ async function run() {
         tl.debug('Checking for ruby install...');
         tl.which('ruby', true);
 
+        //Whenever a specific version of fastlane is requested, we're going to uninstall all installed
+        //versions of fastlane beforehand.  Note that this doesn't uninstall dependencies of fastlane.
+        if (installFastlane && fastlaneVersionToInstall) {
+            let gemRunner: ToolRunner = tl.tool(tl.which('gem', true));
+            gemRunner.arg(['uninstall', 'fastlane']);
+            tl.debug(`Uninstalling all fastlane versions...`);
+            gemRunner.arg(['-a', '-I']);  //uninstall all versions
+            await gemRunner.exec();
+        }
         // If desired, install the fastlane tools (if they're already present, should be a no-op)
         if (installFastlane) {
             tl.debug('Installing fastlane...');
@@ -141,7 +150,7 @@ async function run() {
         } else if (releaseTrack === 'Production') {
             let bundleIdentifier: string = tl.getInput('appIdentifier', true);
             // Run deliver (via fastlane) to publish to Production track
-            // See https://github.com/fastlane/deliver for more information on these arguments
+            // See https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/options.rb for more information on these arguments
             let deliverCommand: ToolRunner = tl.tool('fastlane');
             deliverCommand.arg(['deliver', '--force', '-u', credentials.username, '-a', bundleIdentifier, '-i', ipaPath]);
             deliverCommand.argIf(skipBinaryUpload, ['--skip_binary_upload', 'true']);
@@ -157,8 +166,8 @@ async function run() {
             } else {
                 deliverCommand.arg(['--skip_screenshots', 'true']);
             }
-            deliverCommand.argIf(teamId, ['-q', teamId]);
-            deliverCommand.argIf(teamName, ['-r', teamName]);
+            deliverCommand.argIf(teamId, ['-k', teamId]);
+            deliverCommand.argIf(teamName, ['-e', teamName]);
             deliverCommand.argIf(shouldSubmitForReview, ['--submit_for_review', 'true']);
             deliverCommand.argIf(shouldAutoRelease, ['--automatic_release', 'true']);
             await deliverCommand.exec();

--- a/Tasks/app-store-release/task.json
+++ b/Tasks/app-store-release/task.json
@@ -13,7 +13,7 @@
     "demands": [ "xcode" ],
     "version": {
         "Major": "1",
-        "Minor": "114",
+        "Minor": "115",
         "Patch": "0"
     },
     "minimumAgentVersion": "1.95.3",

--- a/Tasks/app-store-release/task.json
+++ b/Tasks/app-store-release/task.json
@@ -243,7 +243,7 @@
             "defaultValue": "",
             "required": true,
             "groupName": "advanced",
-            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).",
+            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
             "visibleRule": "fastlaneToolsVersion = SpecificVersion"
         }
     ],

--- a/Tasks/app-store-release/task.loc.json
+++ b/Tasks/app-store-release/task.loc.json
@@ -15,7 +15,7 @@
   ],
   "version": {
     "Major": "1",
-    "Minor": "114",
+    "Minor": "115",
     "Patch": "0"
   },
   "minimumAgentVersion": "1.95.3",

--- a/Tasks/ipa-resign/Strings/resources.resjson/en-US/resources.resjson
+++ b/Tasks/ipa-resign/Strings/resources.resjson/en-US/resources.resjson
@@ -38,7 +38,7 @@
   "loc.input.label.fastlaneToolsVersion": "fastlane Version",
   "loc.input.help.fastlaneToolsVersion": "Choose to install either the lastest version of fastlane or a specific version.",
   "loc.input.label.fastlaneToolsSpecificVersion": "fastlane Specific Version",
-  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).",
+  "loc.input.help.fastlaneToolsSpecificVersion": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
   "loc.messages.DarwinOnly": "The Ipa Resign task can only run on a Mac computer.",
   "loc.messages.SuccessfullyResigned": "Successfully resigned ipa %s",
   "loc.messages.FailedTemporaryKeyDeletion": "Failed to delete temporary keychain created during the resign process. %s",

--- a/Tasks/ipa-resign/ipa-resign.ts
+++ b/Tasks/ipa-resign/ipa-resign.ts
@@ -151,6 +151,15 @@ async function run() {
         tl.debug('Checking for ruby install...');
         tl.which('ruby', true);
 
+        //Whenever a specific version of fastlane is requested, we're going to uninstall all installed
+        //versions of fastlane beforehand.  Note that this doesn't uninstall dependencies of fastlane.
+        if (installFastlane && fastlaneVersionToInstall) {
+            let gemRunner: ToolRunner = tl.tool(tl.which('gem', true));
+            gemRunner.arg(['uninstall', 'fastlane']);
+            tl.debug(`Uninstalling all fastlane versions...`);
+            gemRunner.arg(['-a', '-I']);  //uninstall all versions
+            await gemRunner.exec();
+        }
         // If desired, install the fastlane tools (if they're already present, should be a no-op)
         if (installFastlane) {
             tl.debug('Installing fastlane...');

--- a/Tasks/ipa-resign/task.json
+++ b/Tasks/ipa-resign/task.json
@@ -197,7 +197,7 @@
             "defaultValue": "",
             "required": true,
             "groupName": "advanced",
-            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).",
+            "helpMarkDown": "Provide the version of fastlane to install (e.g., 2.15.1).  If a specific version of fastlane is installed, all previously installed versions will be uninstalled beforehand.",
             "visibleRule": "fastlaneToolsVersion = SpecificVersion"
         }
     ],

--- a/Tasks/ipa-resign/task.json
+++ b/Tasks/ipa-resign/task.json
@@ -12,7 +12,7 @@
     "demands": [ "xcode" ],
     "version": {
         "Major": "1",
-        "Minor": "114",
+        "Minor": "115",
         "Patch": "0"
     },
     "minimumAgentVersion": "1.95.3",

--- a/Tasks/ipa-resign/task.loc.json
+++ b/Tasks/ipa-resign/task.loc.json
@@ -14,7 +14,7 @@
   ],
   "version": {
     "Major": "1",
-    "Minor": "114",
+    "Minor": "115",
     "Patch": "0"
   },
   "minimumAgentVersion": "1.95.3",

--- a/app-store-vsts-extension.json
+++ b/app-store-vsts-extension.json
@@ -2,7 +2,7 @@
     "manifestVersion": 1,
     "id": "app-store",
     "name": "Apple App Store",
-    "version": "1.114.0",
+    "version": "1.115.0",
     "publisher": "ms-vsclient",
     "description": "Provides tasks for publishing to Apple's App Store from a TFS/Team Services build or release definition",
     "categories": [

